### ISSUE TITLE
BUG: modeline-parse not working (?)

### ISSUE THREAD

**Issue by sdothum** [kakoune-lsp/kakoune-lsp#713]
I have in my kakrc file

`hook global BufOpenFile .* %{ modeline-parse }`

"Some" shell scripts I have with

`# kak: filetype=sh`

do not have their filetype set. Manually running "modeline-parse" also does not set the filetype. Running

`set buffer filetype sh`

manually does set the filetype for these problem files.

Even if i remove the few community plugs from autoload and run kak without a kakrc file, the same  results occur for the problem scripts.

Oddly, this is not pervasive. Most of my scripts have their filetype set properly -- i am uncertain whether this is simply file.kak determining the correct filetype. There are no modeline-parse errors in *debug*.

I have to think there is something particular to my setup since i can't imagine this problem has gone unnoticed. Running kakoune on Alpine 3.19 (latest stable).

**Comment by sdothum** [comment]
Found it! Pretty obscure..

modeline-parse was parsing out on nearby lines above (i have my modeline at the end of the file) keywords with a trailing colon ":" (which were neither vim: or kak:) and failing out.

Since i have my modelines at the EOF, i modded modeline.kak and set "modelines" to 2 lines from 5.

**Comment by krobelus** [comment]
> Shouldn't modeline.kak simply ignore anything but lines with kak: or vim: ?

yes and it does as far as I can tell. If you can share an input file that triggers the problem, then it should be easy to fix
